roles/kafka_connect/tasks/main.yml

```diff
@@ -163,21 +175,11 @@
     secrets_file: "{{ kafka_connect_secrets_protection_file }}"
     secrets_file_owner: "{{ kafka_connect_user }}"
     secrets_file_group: "{{ kafka_connect_group }}"
+    ca_cert_path: "{{ kafka_connect_ca_cert_path if kafka_connect_ssl_enabled|bool else '' }}"
     handler: restart connect distributed
-  when: kafka_connect_secrets_protection_enabled|bool
-
-- name: Create Kafka Connect Config
-  template:
-    src: connect-distributed.properties.j2
-    dest: "{{kafka_connect.config_file}}"
-    mode: 0640
-    owner: "{{kafka_connect_user}}"
-    group: "{{kafka_connect_group}}"
-  notify: restart connect distributed
-  when: not kafka_connect_secrets_protection_enabled|bool
-  diff: "{{ not mask_sensitive_diff|bool }}"
-  tags:
-    - configuration
+  when:
+    - kafka_connect_secrets_protection_enabled|bool
+    - rbac_enabled|bool
```

Comment:
This is intentional breaking change in [Confluent Platform 7.1.x](https://docs.confluent.io/confluent-cli/current/migrate.html#breaking-changes-for-confluent-cli) 
This change in Ansible is to align the CLI and behaviour along with the default CLI getting packaged with CP 7.1.0. 
If we put the conditional check for cli version, 
  1) we will diverge from what's expected from CP 7.1.x user 
  2) It may lead to compatibility issues. 
  3) User may compromise on security. The primary reason for the breaking changes was to strengthen the security. 
  
Starting CP 7.1.x, the encryption algorithm resides in the MDS and CLI need to login to MDS to get the algorithm for encryption /decryption. Its doesn't require full fledged RBAC implementation, but s simple implementation of RBAC with only one admin level user will also server the purpose. 
Said that, we have discussed this possible upgrade failure ( secret protection without rbac ).  User would see the failure if they try to *re-encrypt* the configuration while upgrade. If we see there are significant number of such users, we may also add a simple RBAC configuration only for secret protection. 
We will update the document with this breaking change and limitations, as product security would be our primary concern. 
@rohit2b